### ISSUE TITLE
Refactor the binding resolving logic in function plugin

### ISSUE THREAD

**Issue by jdneo** [microsoft/azure-maven-plugins#583]

**Comment by pragnagopa** [comment]
@jdneo / @Flanker32  -  Assuming this issue is to track refactoring code to get versioned templates from https://github.com/Azure/azure-functions-tooling-feed . If not, I will open a separate issue.

@soninaren - Please fill in the details on how maven plugin get the bindings.json from the tooling feed.

cc: @jeffhollan @mattchenderson

**Comment by kulkarnisonia16** [comment]
Waiting on https://github.com/Azure/azure-functions-java-library/issues/69
